filesystem/File.php

```diff
@@ -9,97 +14,120 @@
+ *   is linked to Name, so Name and Title will always be the same.
+ * -"File": Physical asset backing this DB record. This is a composite DB field with
+ *   its own list of properties. {@see DBFile} for more information
+ * -"Content": Typically unused, but handy for a textual representation of
+ *   files, e.g. for fulltext indexing of PDF documents.
+ * -"ParentID": Points to a {@link Folder} record. Should be in sync with
+ *  "Filename". A ParentID=0 value points to the"assets/" folder, not the webroot.
+ * -"ShowInSearch": True if this file is searchable
  *
  * @package framework
  * @subpackage filesystem
  *
- * @property string Name Basename of the file
- * @property string Title Title of the file
- * @property string Filename Filename including path
- * @property string Content
- * @property string ShowInSearch Boolean that indicates if file is shown in search. Doesn't apply to Folder
- *
- * @property int ParentID ID of parent File/Folder
- * @property int OwnerID ID of Member who owns the file
+ * @property string $Name Basename of the file
+ * @property string $Title Title of the file
+ * @property DBFile $File asset stored behind this File record
+ * @property string $Content
+ * @property string $ShowInSearch Boolean that indicates if file is shown in search. Doesn't apply to Folders
+ * @property int $ParentID ID of parent File/Folder
+ * @property int $OwnerID ID of Member who owns the file
  *
  * @method File Parent() Returns parent File
  * @method Member Owner() Returns Member object of file owner.
  */
-class File extends DataObject {
+class File extends DataObject implements ShortcodeHandler, AssetContainer {
 
-	private static $default_sort = "\"Name\"";
+	use ImageManipulation;
 
-	private static $singular_name = "File";
+	/**
+	 * Represents the audio file category
+	 * @var string
+	 */
+	const CATEGORY_AUDIO = 'audio';
 
-	private static $plural_name = "Files";
+	/**
+	 * Represents the archive file category
+	 * @var string
+	 */
+	const CATEGORY_ARCHIVE = 'archive';
+
+	/**
+	 * Represents the document file category
+	 * @var string
+	 */
+	const CATEGORY_DOCUMENT = 'document';
+
+	/**
+	 * Represents the flash file category
+	 * @var string
+	 */
+	const CATEGORY_FLASH = 'flash';
+
+	/**
+	 * Represents the image file category
+	 * @var string
+	 */
+	const CATEGORY_IMAGE = 'image';
+
+	/**
+	 * Represents the supported image category
+	 * @var string
+	 */
+	const CATEGORY_IMAGE_SUPPORTED = 'image/supported';
+
+	/**
+	 * Represents the video file category
+	 * @var string
+	 */
+	const CATEGORY_VIDEO = 'video';
+
+	private static $default_sort ="\"Name\"";
+
+	private static $singular_name ="File";
+
+	private static $plural_name ="Files";
 
 	private static $db = array(
-		"Name" => "Varchar(255)",
-		"Title" => "Varchar(255)",
-		"Filename" => "Text",
-		"Content" => "Text",
+		"Name" =>"Varchar(255)",
+		"Title" =>"Varchar(255)",
+		"File" =>"DBFile",
+		"Content" =>"Text",
```

Comment:
Probably a good time to remove the `File.Content` field? Its not exposed via the CMS form, and stays unpopulated unless you write custom functionality - so quite confusing to newcomers. The [textextraction](https://github.com/silverstripe-labs/silverstripe-textextraction) module uses a separate `File.FileContentCache` field as well.